File: app/src/main/java/parkhomov/andrew/lensthicknesscalculator/activities/fragment/glossary/GlossaryDetails.java
package parkhomov.andrew.lensthicknesscalculator.activities.fragment.glossary;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.content.ContextCompat;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import parkhomov.andrew.lensthicknesscalculator.R;
import parkhomov.andrew.lensthicknesscalculator.activities.fragment.Parent;
import parkhomov.andrew.lensthicknesscalculator.activities.utils.CONSTANT;
import parkhomov.andrew.lensthicknesscalculator.activities.utils.Utils;

/**
 * Created by MyPC on 29.07.2017.
 */

public class GlossaryDetails extends Parent {

    @BindView(R.id.glossaryImageView)
    ImageView imageHolder;
    @BindView(R.id.glossaryTitleTextView)
    TextView titleHolder;
    @BindView(R.id.glossaryDescriptionTextView)
    TextView descriptionHolder;
    @BindView(R.id.header)
    TextView header;

    private String title, description;
    private int imageId;

    public static GlossaryDetails getInstance(String title, String desrciption, Integer imageId) {
        Bundle bundle = new Bundle();
        GlossaryDetails glossaryDetails = new GlossaryDetails();
        glossaryDetails.setArguments(bundle);
        glossaryDetails.setTitle(title);
        glossaryDetails.setDescription(desrciption);
        glossaryDetails.setImageId(imageId);
        return glossaryDetails;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setImageId(int imageId) {
        this.imageId = imageId;
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.glossary_detail, container, false);
        ButterKnife.bind(this, view);

        header.setText(Utils.spacing(title.toUpperCase(), CONSTANT.FRAGMENT_HEADER_SPACING_DISTANCE_0_8));
        //set data in holder
        imageHolder.setImageDrawable(ContextCompat.getDrawable(getActivity(), imageId));
        titleHolder.setText(title);
        descriptionHolder.setText(description);

        return view;
    }

    @OnClick(R.id.turnBackImgB)
    public void onBackPressed() {
        getFragmentManager().popBackStack();
    }

}

File: app/src/main/java/parkhomov/andrew/lensthicknesscalculator/fragment/glossary/GlossaryDetails.kt
package parkhomov.andrew.lensthicknesscalculator.fragment.glossary

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.content.ContextCompat
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import butterknife.BindView
import butterknife.ButterKnife
import butterknife.OnClick
import parkhomov.andrew.lensthicknesscalculator.R
import parkhomov.andrew.lensthicknesscalculator.utils.CONSTANT
import parkhomov.andrew.lensthicknesscalculator.utils.Utils

/**
 * Created by MyPC on 29.07.2017.
 */

class GlossaryDetails : Fragment() {

    @BindView(R.id.glossaryImageView)
    lateinit var imageHolder: ImageView
    @BindView(R.id.glossaryTitleTextView)
    lateinit var titleHolder: TextView
    @BindView(R.id.glossaryDescriptionTextView)
    lateinit var descriptionHolder: TextView
    @BindView(R.id.header)
    lateinit var header: TextView

    private var title: String? = null
    private var description: String? = null
    private var imageId: Int = 0

    fun setTitle(title: String) {
        this.title = title
    }

    fun setDescription(description: String) {
        this.description = description
    }

    fun setImageId(imageId: Int) {
        this.imageId = imageId
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater!!.inflate(R.layout.glossary_detail, container, false)
        ButterKnife.bind(this, view)

        header.text = Utils.spacing(title!!.toUpperCase(), CONSTANT.FRAGMENT_HEADER_SPACING_DISTANCE_0_8)
        header.ellipsize = TextUtils.TruncateAt.MARQUEE
        header.marqueeRepeatLimit = -1
        header.setSingleLine(true)
        header.isSelected = true
        //set data in holder
        imageHolder.setImageDrawable(ContextCompat.getDrawable(activity, imageId))
        titleHolder.text = title
        descriptionHolder.text = description

        return view
    }

    @OnClick(R.id.turnBackImgB)
    fun onBackPressed() {
        fragmentManager.popBackStack()
    }

    companion object {

        fun getInstance(title: String, description: String, imageId: Int): GlossaryDetails {
            val bundle = Bundle()
            val glossaryDetails = GlossaryDetails()
            glossaryDetails.arguments = bundle
            glossaryDetails.setTitle(title)
            glossaryDetails.setDescription(description)
            glossaryDetails.setImageId(imageId)
            return glossaryDetails
        }
    }
}
